core/src/main/java/org/apache/carbondata/core/keygenerator/columnar/impl/MultiDimKeyVarLengthEquiSplitGenerator.java

```diff
@@ -143,17 +142,18 @@ private void intialize() {
 
   private int[] convertToArray(List<Integer> list) {
     int[] ints = new int[list.size()];
-    for (int i = 0; i < ints.length; i++) {
-      ints[i] = list.get(i);
+    int i = 0;
+    for (Integer ii : list) {
```

Comment:
just checked online, the performance difference is on List.forEach() as well as stream()..